src/util/cached-prop-defs.js

```diff
@@ -0,0 +1,83 @@
+//@flow
+import assign from './assign';
+import dashCase from './dash-case';
+import getAllKeys from './get-all-keys';
+
+const CACHED_PROP_DEFS:string = '____skate_propDefs';
```

Comment:
And I'm only using Flow internally to help me refactor the code.
For skatejs right now Flow is a better choice.
The syntax is very similar. If you know TypeScript you can learn Flow in a few days.
I used TypeScript for 2 years but now for functional programming I prefer Flow. No extra compiler needed, just annotate your JS or ES6 code. It is closer to the platform.